include/zephyr/mgmt/mcumgr/transport/smp.h

```diff
@@ -134,6 +134,22 @@ struct smp_transport {
 #endif
 };
 
+/**
+ * @brief SMP transport type for client registration
+ */
+enum smp_transport_type {
+	/** SMP serial */
+	SMP_SERIAL_TRANSPORT = 0,
+	/** SMP bluetooth */
+	SMP_BLUETOOTH_TRANSPORT,
+	/** SMP UDP IPv4 */
+	SMP_UDP_IPV4_TRANSPORT,
+	/** SMP UDP IPv6 */
+	SMP_UDP_IPV6_TRANSPORT
+};
+
+#define SMP_CLIENT_TRANPORT_MAX 4
```

Comment:
It is define how many Transport can be register for client.

Reply:
So how is this usable to anyone with an out of zephyr tree transport?

Reply:
Ah OK i understand it is not used outside. Let me move this to internal usage only.

Reply:
MCUmgr is extensible, anyone can add a custom transport and that has worked since MCUmgr was added to zephyr, if you are saying there is a limitation of the client not being able to support that then we need to revisit if this client can be added at all because that is a huge limitation and hiding it/making it internal usage only does not resolve that issue. A user should be able to add their own transport, and use it in client mode if they so wish.

Reply:
Maybe I to transport register by linked list. So Transport define locally structure which is added to linked list. Then Client side is not limit anything.